dask/array/random.py

```diff
@@ -20,6 +20,604 @@
+    def standard_exponential(self, size=None, chunks="auto", **kwargs):
+        return self._wrap("standard_exponential", size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def standard_gamma(self, shape, size=None, chunks="auto", **kwargs):
+        return self._wrap("standard_gamma", shape, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def standard_normal(self, size=None, chunks="auto", **kwargs):
+        return self._wrap("standard_normal", size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def standard_t(self, df, size=None, chunks="auto", **kwargs):
+        return self._wrap("standard_t", df, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def triangular(self, left, mode, right, size=None, chunks="auto", **kwargs):
+        return self._wrap(
+            "triangular", left, mode, right, size=size, chunks=chunks, **kwargs
+        )
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def uniform(self, low=0.0, high=1.0, size=None, chunks="auto", **kwargs):
+        return self._wrap("uniform", low, high, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def vonmises(self, mu, kappa, size=None, chunks="auto", **kwargs):
+        return self._wrap("vonmises", mu, kappa, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def wald(self, mean, scale, size=None, chunks="auto", **kwargs):
+        return self._wrap("wald", mean, scale, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def weibull(self, a, size=None, chunks="auto", **kwargs):
+        return self._wrap("weibull", a, size=size, chunks=chunks, **kwargs)
+
+    @derived_from(np.random.Generator, skipblocks=1)
+    def zipf(self, a, size=None, chunks="auto", **kwargs):
+        return self._wrap("zipf", a, size=size, chunks=chunks, **kwargs)
+
+
+def _choice_rng(state_data, a, size, replace, p, axis, shuffle):
+    state = np.random.default_rng(state_data)
+    return state.choice(a, size=size, replace=replace, p=p, axis=axis, shuffle=shuffle)
+
+
+def _shuffle(state_data, x, axis=0):
+    bit_generator = np.random.PCG64()
+    bit_generator.state = state_data
+    state = np.random.Generator(bit_generator)
+    return state.shuffle(x, axis=axis)
+
+
+def _spawn_bitgens(bitgen, n_bitgens):
+    seeds = bitgen._seed_seq.spawn(n_bitgens)
+    bitgens = [type(bitgen)(seed) for seed in seeds]
+    return bitgens
+
+
+def _apply_random_func(rng, funcname, bitgen, size, args, kwargs):
+    """Apply random method with seed"""
+    if rng is None or rng is Generator:
+        state = np.random.default_rng(bitgen)
+    else:
+        # cupy or cupy-like library that provides `random.default_rng()`
+        import importlib
```

Comment:
Would suggest moving this `import` to the top of the module

Reply:
done [c82f15](https://github.com/dask/dask/pull/9038/commits/c82f15e54266e9ba8814f2d7a3727d14bd32830a)